### PR TITLE
Add gateway alerts to Monit, for #2919

### PR DESCRIPTION
**Pull request by MichaelDeciso**

### CODE REVIEW

File: src/etc/inc/gwlb.inc [hunk 1]
```diff
@@ -30,6 +30,10 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+require_once 'interfaces.inc';
+require_once 'interfaces.lib.inc';
+require_once 'util.inc';
```
Comment (AdSchellevis) [inline src/etc/inc/gwlb.inc]: remove requires

File: src/etc/inc/gwlb.inc [hunk 2]
```diff
@@ -993,3 +979,22 @@ function running_dpinger_processes()
 
     return $result;
 }
+
```
Comment (AdSchellevis) [inline src/etc/inc/gwlb.inc]: better to move `get_gateway_error()` to the monit script itself, it's not used elsewhere.

File: src/opnsense/scripts/OPNsense/Monit/gateway_alert [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+require_once 'config.inc';
+require_once 'gwlb.inc';
```
Comment (AdSchellevis) [inline src/opnsense/scripts/OPNsense/Monit/gateway_alert]: for consistency better to wrap in brackets

File: src/etc/inc/gwlb.inc [hunk 4]
```diff
@@ -993,3 +975,22 @@ function running_dpinger_processes()
...
+    if (stristr($status, 'down') !== false) {
+        return sprintf(gettext('MONITOR: %s is down, removing from routing group %s'), $gwname, $group['name']);
+    } elseif (stristr($status, 'loss') !== false && stristr($group['trigger'], 'loss') !== false) {
+        return sprintf(gettext('MONITOR: %s has packet loss, removing from routing group %s'), $gwname, $group['name']);
+    } elseif (stristr($status, 'delay') !== false && stristr($group['trigger'], 'latency') !== false) {
+        return sprintf(gettext('MONITOR: %s has high latency, removing from routing group %s'), $gwname, $group['name']);
+    } else {
+        return null;
+    }
+}
```
Comment (fichtner) [inline src/etc/inc/gwlb.inc]: No newline at end of file, usually the following script will clean up all whitespace lint...

    make sweep

File: src/opnsense/mvc/app/models/OPNsense/Monit/Migrations/M1_0_7.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\Monit\Migrations;
+
+use OPNsense\Base\BaseModelMigration;
+
+class M1_0_7 extends BaseModelMigration
```
Comment (fichtner) [inline src/opnsense/mvc/app/models/OPNsense/Monit/Migrations/M1_0_7.php]: In general, shouldn't we be starting to use a "plugin" directory to bring these scripts in and avoid the model migration?

Comment (AdSchellevis) [reply src/opnsense/mvc/app/models/OPNsense/Monit/Migrations/M1_0_7.php]: @fichtner I think we should make sure the current construction can function first, doing this properly providing backward compatibility will be quite difficult (if possible). 

We'll see where we end up, original goal was to replace the notices with Monit, let's start improving what there is now step by step.

Comment (fichtner) [reply src/opnsense/mvc/app/models/OPNsense/Monit/Migrations/M1_0_7.php]: Ok, ageeed

**Pull request merged**